packages/core/state/selection/selectors.ts

```diff
@@ -59,3 +61,19 @@ export const getAnnotationFilters = createSelector([getFileFilters], (fileFilter
 export const getFileAttributeFilter = createSelector([getFileFilters], (fileFilters):
     | FileFilter
     | undefined => fileFilters.find((f) => TOP_LEVEL_FILE_ANNOTATION_NAMES.includes(f.name)));
+
+export const getGroupedByFilterName = createSelector(
+    [getAnnotationFilters, getAnnotations],
+    (globalFilters: FileFilter[], annotations: Annotation[]) => {
+        const annotationNameToInstanceMap = keyBy(annotations, "name");
+        const filters: Filter[] = map(globalFilters, (filter: FileFilter) => {
+            const annotation = annotationNameToInstanceMap[filter.name];
+            return {
+                name: filter.name,
+                value: filter.value,
+                displayValue: annotation?.getDisplayValue(filter.value),
+            };
+        }).filter((filter) => filter.displayValue !== undefined);
+        return groupBy(filters, (filter) => filter.name);
+    }
+);
```

Comment:
I like the createSelector solution! The only useMemo approach that I have thought of for this would be either to move the useMemo call up in the component hierarchy to a common ancestor of all the places you want to use it. This seems preferable.